### DISCUSSION TITLE
Welcome to Douyin_TikTok_Download_API Discussions!

### DISCUSSION THREAD

**Discussion by Evil0ctal** [Evil0ctal/Douyin_TikTok_Download_API#55] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Share your own API server
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shreateh** [comment]
Hi,
I cant download ios shortcut, loading icon forever
Ios 14.8 , iphone 11 pro max

**Reply by Evil0ctal** [reply]
Please check your network

**Reply by shreateh** [reply]
Thanks for replying.. tried on 2 diffident ISPs, 3G israeli isp, and adsl Palestinian iso, 
bot same issue : https://i.imgur.com/lsZnbzF.png

**Reply by Evil0ctal** [reply]
> Thanks for replying.. tried on 2 diffident ISPs, 3G israeli isp, and adsl Palestinian iso, 
> bot same issue : https://i.imgur.com/lsZnbzF.png 

Can you try use US VPN to download the shortcut? Cause shortcuts are saved by Apple's iCloud server.

**Reply by shreateh** [reply]
hi, just tried on US VPN , also on a new FIBER ISP , same issue .. could you please export it as .shortcut using icloud backup? if that is possible with you. or just provide the end point download URL to build a new one from scratch. 
thanks 
EDIT: 
Thank you, after manual code review i figure out how to build one, this work is amazing . 
is there a script to get video comments?
many thanks :)

**Reply by Evil0ctal** [reply]
> hi, just tried on US VPN , also on a new FIBER ISP , same issue .. could you please export it as .shortcut using icloud backup? if that is possible with you. or just provide the end point download URL to build a new one from scratch. 
> 
> thanks 
> 
> EDIT: 
> 
> Thank you, after manual code review i figure out how to build one, this work is amazing . 
> 
> is there a script to get video comments?
> 
> many thanks :) 

Currently there is no API for getting comments, but I will be able to create one, can you open an issue in my repo? So I won't forget it.

**Reply by Evil0ctal** [reply]
TikTok comments API had been finished.

Link:

https://api-v2.douyin.wtf/docs

**Comment by shawn35** [comment]
hi, is there a api that can check the fastest video release from douyin profiles? example once the profile releasenew video in 1 second, i can check it

**Reply by Evil0ctal** [reply]
Currently there is an idea can do that, but 1 second is too fast, if you requested TikTok server every second,they will block your IP address, so you need a big proxy pool.

**Reply by shawn35** [reply]
i am thinking is there any way i can download it (within 1-10sec) after video release from douyin and auto upload the video to facebook right manager

**Reply by Evil0ctal** [reply]
Please check this new API:

https://api-v2.douyin.wtf/docs

**Comment by artemmarkov050** [comment]
I did everything as instructed, but I don't really understand how I can find my api, what do I have to do to find a url like you have '[https://douyin.wtf/api?url={video_url}](https://douyin.wtf/api?url=%7Bvideo_url%7D)' ?
![апи тикток](https://github.com/Evil0ctal/Douyin_TikTok_Download_API/assets/99012014/567bad30-553a-4007-b1c9-3dfaebe95c95)

**Reply by Evil0ctal** [reply]
YourIP:Port/api

**Comment by androidformax** [comment]
If I try get douyin link, server ansewer - Internal Server Error buit tiktok is work, have you any issue?
